cluster/examples/kubernetes/rook-operator.yaml

```diff
@@ -122,3 +128,17 @@ spec:
         # current mon with a new mon (useful for compensating flapping network).
         - name: ROOK_MON_OUT_TIMEOUT
           value: "300s"
+        - name: ROOK_AGENT_IMAGE_TAG
+          value: master
```

Comment:
for what scenario would we want the agents to be running a different version than the operator?

Reply:
Good question. i am not sure.. But since we already have ways to specify a different tag for the cluster (like mons, osd, rgw etc) via the `versionTag`, i thought to be consistent and follow this model.

Reply:
yes, this seems necessary. In the upgrade path, the sequence would be:
1. Upgrade the operator. It would start with the new version tag for the agent
2. The operator would upgrade the agents
3. The user would update the cluster specs
4. The operator would update the daemons according to that desired version. The operator could also enforce that the agent be at least the same version as the cluster before it upgrades.

Reply:
The reason i added it was to remain consistent with how we do for other resources that the operator spawns. For example, for cluster, we are also specifying the tag version. I can remove it as i dont have any reasoning to have the agent version tag differ the operator

Reply:
ok.. ill remove it.